src/xunit.console.uwp/HostedAppxTest.cs

```diff
@@ -132,27 +140,28 @@ public void Run(bool debug, TimeSpan timeout)
             }
             IntPtr pid;
             Console.WriteLine("Activating: " + appUserModelId);
-            var hri = activationManager.ActivateApplication(appUserModelId, this.argsToPass, ACTIVATEOPTIONS.AO_NOERRORUI | ACTIVATEOPTIONS.AO_NOSPLASHSCREEN, out pid);
+            var hr = activationManager.ActivateApplication(appUserModelId, this.argsToPass, ACTIVATEOPTIONS.AO_NOERRORUI | ACTIVATEOPTIONS.AO_NOSPLASHSCREEN, out pid);
             var timer = Stopwatch.StartNew();
-            Console.WriteLine("UWP Install HRESULT: " + hri);
-            var p = Process.GetProcessById(pid.ToInt32());
-            Console.WriteLine("Running {0} in process {1} at {2}", p.ProcessName, pid, DateTimeOffset.Now);
-            while (timer.ElapsedMilliseconds < timeout.TotalMilliseconds && !p.HasExited)
-            {
-                Thread.Sleep(1000);
-            }
-            var cleanExit = p.HasExited;
-            if (!p.HasExited)
+            Console.WriteLine("UWP Activation HRESULT: " + hr);
+            if (hr == 0)
             {
-                Console.WriteLine("Killing {0}", pid);
-                p.Kill();
+                var p = Process.GetProcessById(pid.ToInt32());
+                Console.WriteLine("Running {0} in process {1} at {2}", p.ProcessName, pid, DateTimeOffset.Now);
+                while (timer.ElapsedMilliseconds < timeout.TotalMilliseconds && !p.HasExited)
+                {
+                    Thread.Sleep(1000);
+                }
+                var cleanExit = p.HasExited;
+                if (!p.HasExited)
```

Comment:
Should use cleanExit here.  It's possible for the property's value to change between these two lines resulting in saying "I'm killing this, and it cleanly exited!"